src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,917 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+
+    try:
+        with open(config_file_location, 'w') as f:
+            yaml.dump(dict_file, f, default_flow_style=False)
+    except Exception as e:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=e, fault_type=consts.Create_Config_Fault_Type,
+                                summary='Unable to create config file for proxy.')
+        raise CLIError("Failed to create config for proxy." + str(e))
+
+    args.append("-c")
+    args.append(config_file_location)
+
+    debug_mode = False
+    if '--debug' in cmd.cli_ctx.data['safe_params']:
+        args.append("-d")
+        debug_mode = True
+
+    client_side_proxy(cmd, client, resource_group_name, cluster_name, 0, args, client_proxy_port, api_server_port, operating_system, creds, user_type, debug_mode, token=token, path=path, overwrite_existing=overwrite_existing, context_name=context_name)
+
+
+# Prepare data as needed by client proxy executable
+def prepare_clientproxy_data(response):
+    data = {}
+    data['kubeconfigs'] = []
+    kubeconfig = {}
+    kubeconfig['name'] = 'Kubeconfig'
+    kubeconfig['value'] = b64encode(response.kubeconfigs[0].value).decode("utf-8")
+    data['kubeconfigs'].append(kubeconfig)
+    data['hybridConnectionConfig'] = {}
+    data['hybridConnectionConfig']['relay'] = response.hybrid_connection_config.relay
+    data['hybridConnectionConfig']['hybridConnectionName'] = response.hybrid_connection_config.hybrid_connection_name
+    data['hybridConnectionConfig']['token'] = response.hybrid_connection_config.token
+    data['hybridConnectionConfig']['expirationTime'] = response.hybrid_connection_config.expiration_time
+    return data
+
+
+def client_side_proxy(cmd,
+                      client,
+                      resource_group_name,
+                      cluster_name,
+                      flag,
+                      args,
+                      client_proxy_port,
+                      api_server_port,
+                      operating_system,
+                      creds,
+                      user_type,
+                      debug_mode,
+                      token=None,
+                      path=os.path.join(os.path.expanduser('~'), '.kube', 'config'),
+                      overwrite_existing=False,
+                      context_name=None):
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    if token is not None:
+        auth_method = 'Token'
+        telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.IsAADEnabled': False})
+    else:
+        auth_method = 'AAD'
+        telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.IsAADEnabled': True})
+
+    # Fetching hybrid connection details from Userrp
+    try:
+        response = client.list_cluster_user_credentials(resource_group_name, cluster_name, auth_method, True)
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Get_Credentials_Failed_Fault_Type,
```

Comment:
Use utils.arm_exception_handler here

Reply:
yeah, I have replaced it with this method.